app/utility/base_planning_svc.py

```diff
@@ -113,6 +118,39 @@ async def obfuscate_commands(self, agent, obfuscator, links):
 
     """ PRIVATE """
 
+    @staticmethod
+    def _fil(link, trait):
+        """
+        Filter Link based on presence of trait in facts used
+        """
+        if any(trait in y.trait for y in link.used):
+            return link
+
+    @staticmethod
+    def _list_historic_dup_lat_mov(operation):
+        lat = [BasePlanningService._fil(k, 'remote.host.fqdn') for k in operation.chain if k.status != 1]
+        return [(x.command_hash if x.command_hash else x.command) for x in lat if x]
+
+    @staticmethod
+    def _cross_check_agents_for_duplicate_lat_mov(agent_links):
+        """
+        Filter links across agents
+        :param agent_links: array of agent links
+        :return: Flattened, filtered list of links
+        """
+        links = []
+        parallel_list = []
+        for agent_list in agent_links:
+            for el in agent_list:
+                if not BasePlanningService._fil(el, 'remote.host.fqdn'):
```

Comment:
I'm not sure how I feel about us statically defining a case for a fact within the base planning service but this will probably be something that'll have to be taken care of with the overall fact engine rework.

Reply:
Additionally, is there any other way we can do this? What if there are custom lateral movement abilities that don't rely on remote.host.fqdn? or if the FQDN fact is different for a machine that has 2 NICs for example, etc

Reply:
I see this a patch for now, not a permanent solution, as this is clearly a very specialized operation/use-case.

I like the half-backed ideas we have had on using locks/mutexs for fact/trait/value usage restrictions which would allow a more general mechanism for enforcing rules.

Reply:
The locks/mutex stuff is likely the best solution for this, but that's a refactor in the future.